
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Welcome to SiFoN’s documentation! &#8212; SiFoN 0.0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/classic.css" />
    
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="nav-item nav-item-0"><a href="#">SiFoN 0.0.1 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Welcome to SiFoN’s documentation!</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="welcome-to-sifon-s-documentation">
<h1>Welcome to SiFoN’s documentation!<a class="headerlink" href="#welcome-to-sifon-s-documentation" title="Permalink to this headline">¶</a></h1>
</section>
<section id="module-SiFoN.chrom_viz">
<span id="chrom-viz-api"></span><h1>chrom_viz API<a class="headerlink" href="#module-SiFoN.chrom_viz" title="Permalink to this headline">¶</a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.chrom_viz.chromatin_profile_heatmap">
<span class="sig-prename descclassname"><span class="pre">SiFoN.chrom_viz.</span></span><span class="sig-name descname"><span class="pre">chromatin_profile_heatmap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row_labels</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc_index</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figname</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_X</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_window</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#SiFoN.chrom_viz.chromatin_profile_heatmap" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates, displays, and saves a heatmap of the top chromatin profile scores within a short genomic sequence. The y axis shows the top chromatin profiles for each of the user-defined SNPs of interest. The profile labels are in the following format: chromatin profile | tissue of origin | genomic position of associated SNP. The x axis shows genomic positions. 
:param data: NumPy array of Sei chromatin profile scores. Should have shape (X, 21907), where 21,907 corresponds to the number of chromatin profiles and X corresponds to the number of alterations.
:type data: NumPy array
:param row_labels: contains metadata to describe the SNPs being analyzed. Should have at least the following columns: [“chrom”, “pos”, “name”, “ref”, “alt”]. Must have the same number of rows as <cite>data</cite>. 
:type row_labels: Pandas DataFrame
:param loc: the chromosomal position of the SNP of interest.
:type loc: int
:param loc_index: the index of the SNP of interest in the <cite>data</cite> array.
:type loc_index: int
:param figname: name of file that figure will be saves as.
:type figname: string
:param top_X: the number of top scoring chromatin profiles to plot for each SNP. Increasing top_X increases the height of
plot. There will be <cite>top_X</cite> * <cite>pos_window</cite> rows and pos_window columns, defaults to 5.
:type top_X: int, optional
:param window: the number of to the right and left of <cite>loc</cite> that will be plotted, defaults to 5
:type window: int, optional
:param fontsize: fontsize for graph, defaults to 8
:type fontsize: int, optional.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.chrom_viz.preprocess_diff">
<span class="sig-prename descclassname"><span class="pre">SiFoN.chrom_viz.</span></span><span class="sig-name descname"><span class="pre">preprocess_diff</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">diff</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row_labels</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#SiFoN.chrom_viz.preprocess_diff" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a filtered chromatin class score array and corresponding row labels. For any particular genomic loci (e.g.chr10:89623103), Sei outputs three scores corresponding to each potential alteration at that position (e.g. G&gt;A, G&gt;C, and G&gt;T). For each position, this function selects the alteration that has the highest average absolute score across all chromatin classes. 
:param diff: NumPy array of Sei chromatin profile scores. Should have shape (X, 21907), where 21,907 corresponds to the number of chromatin profiles and X corresponds to the number of alterations.
:type diff: NumPy array
:param row_labels: contains metadata about the rows (and the SNPs described by the rows) in the <cite>diff</cite> array. 
:type row_labels: Pandas DataFrame
:return: Returns filtered <cite>diff</cite> and <cite>row_labels</cite>
:rtype: NumPy array, Pandas DataFrame</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.chrom_viz.rank_scatter_plot">
<span class="sig-prename descclassname"><span class="pre">SiFoN.chrom_viz.</span></span><span class="sig-name descname"><span class="pre">rank_scatter_plot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">diff</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row_labels</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc_index</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figname</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">18</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">static</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#SiFoN.chrom_viz.rank_scatter_plot" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates, displays, and saves a scatter plot of ranked chromatin profiles at for a particular SNP. Each point corresponds to a chromatin profile, with hover data specifying: rank, score, profile name, and tissue of origin.
:param diff: NumPy array of Sei chromatin profile scores. Should have shape (X, 21907), where 21,907 corresponds to the number of chromatin profiles and X corresponds to the number of alterations.
:type diff: NumPy array
:param row_labels: contains metadata to describe the SNPs being analyzed. Should have at least the following columns: [“chrom”, “pos”, “name”, “ref”, “alt”]. Must have the same number of rows as <cite>data</cite>. 
:type row_labels: Pandas DataFrame
:param loc: the chromosomal position of the SNP of interest.
:type loc: int
:param loc_index: the index of the SNP of interest in the <cite>data</cite> array.
:type loc_index: int
:param figname: name of file that figure will be saves as.
:type figname: string
:param fontsize: fontsize for graph, defaults to 8
:type fontsize: int, optional.
:param static: species whether to output a static png image (True) or an interactive html image in your notebook (False), defaults to False
:type boolean, optional</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.chrom_viz.white_bg">
<span class="sig-prename descclassname"><span class="pre">SiFoN.chrom_viz.</span></span><span class="sig-name descname"><span class="pre">white_bg</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fig</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#SiFoN.chrom_viz.white_bg" title="Permalink to this definition">¶</a></dt>
<dd><p>This function changes a plotly figure <cite>fig</cite> to have a white background. Changes the figure object. 
:param fig: plotly figure
:type fig: plotly figure</p>
</dd></dl>

</section>
<section id="module-SiFoN.clinical_analysis">
<span id="clinical-analysis-api"></span><h1>clinical_analysis API<a class="headerlink" href="#module-SiFoN.clinical_analysis" title="Permalink to this headline">¶</a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.clinical_analysis.add_case_control_label">
<span class="sig-prename descclassname"><span class="pre">SiFoN.clinical_analysis.</span></span><span class="sig-name descname"><span class="pre">add_case_control_label</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">case_cuttoff</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">control_cutoff</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#SiFoN.clinical_analysis.add_case_control_label" title="Permalink to this definition">¶</a></dt>
<dd><p>Assigns each SNP as a “Case”, “Control”, or “Equal” based on odds ratio scores. Adds dataframe column “Case/Control” to <cite>df</cite>.
:param df: Contains case and control counts and any other metadata. Must have an “Odds Ratio” column.
:type df: Pandas DataFrame
:param case_cuttoff: cutoff odds ratio to consider something a “Case” SNP
:type case_cuttoff: int
:param control_cutoff: cutoff odds ratio to consider something a “Control” SNP
:type control_cutoff: int</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.clinical_analysis.cdot_to_VCF">
<span class="sig-prename descclassname"><span class="pre">SiFoN.clinical_analysis.</span></span><span class="sig-name descname"><span class="pre">cdot_to_VCF</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrm</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#SiFoN.clinical_analysis.cdot_to_VCF" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts alterations from the c-dot (c. pos ref &gt; alt) format to a VCF file.
:param data: Clinical data. Must contain a column “Alteration” in c. notation that describes SNP’s genomic coordinates.
:type data: Pandas DataFrame
:param chrm: Chromosome of alterations. Should be in the form “chr” + the chromosome number (e.g. chr10).
:type chrm: string
:param start: Reference genomic position for alterations, default is 0
:type start: int, optional</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>Input data stored in a VCF format dataframe.</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>Pandas DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.clinical_analysis.odds_ratio">
<span class="sig-prename descclassname"><span class="pre">SiFoN.clinical_analysis.</span></span><span class="sig-name descname"><span class="pre">odds_ratio</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">control_col</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">case_col</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_cases</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_controls</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">correction</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#SiFoN.clinical_analysis.odds_ratio" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the odds ratio of case and control counts. Adds a new column to your dataframe <cite>df</cite> called “Odds Ratio”
:param df: Contains case and control counts and any other metadata.
:type df: Pandas DataFrame
:param control_col: name of DataFrame column with control counts
:type control_col: string
:param case_col: name of DataFrame column with case counts
:type case_col: string
:param num_cases: number of total cases
:type num_cases: int
:param num_controls: number of total controls
:type num_controls: int
:param correction: added to all counts to prevent division by zero in the case of zero counts.
:type correction: float</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.clinical_analysis.seq_class_t_tests">
<span class="sig-prename descclassname"><span class="pre">SiFoN.clinical_analysis.</span></span><span class="sig-name descname"><span class="pre">seq_class_t_tests</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figname</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fontsize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">18</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">markersize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">7</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#SiFoN.clinical_analysis.seq_class_t_tests" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs t-tests to compare sequence class scores in case and control populations. Returns p-vals and plots ranked p-vals.
:param df: Contains case and control counts and any other metadata. Must have scores for all 40 Sei sequence classes. Column names must correspond to sequence class names. 
:type df: Pandas DataFrame
:param figname: name of file that figure will be saves as.
:type figname: string
:param fontsize: fontsize for graph, defaults to 18
:type fontsize: int, optional.
:param markersize: markersize for graph, defaults to 7
:type markersize: int, optional.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>P-values for all 40 sequence class, ordered by ascending p-value.</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>Pandas DataFrame</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-SiFoN.GWAS_enrichment">
<span id="gwas-enrichment-api"></span><h1>GWAS_enrichment API<a class="headerlink" href="#module-SiFoN.GWAS_enrichment" title="Permalink to this headline">¶</a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.GWAS_enrichment.find_and_plot_enrichment">
<span class="sig-prename descclassname"><span class="pre">SiFoN.GWAS_enrichment.</span></span><span class="sig-name descname"><span class="pre">find_and_plot_enrichment</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ccv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_bed</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figname</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vmin</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vmax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">11</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#SiFoN.GWAS_enrichment.find_and_plot_enrichment" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates sequence class enrichment in genomic sequences compared to the entire genome and then creates a heatmap to visualize results. 
:param ccv: Pandas DataFrame in BED format of loci with credible causal variants. Must include at least the following columns: Chrom, Start, End, SNP [identifier/name of sequence])
:type ccv: Pandas DataFrame
:param ref_bed: Should contain sequence class labels across (nearly) the entire genome to allow for background sequence class proportion calculations. Should use the same genomic coordinate system (e.g. hg19 or hg38) as <cite>ccv</cite>.
:type ref_bed: pyranges BED object
:param figname: name of file that figure will be saves as.
:type figname: string
:param vmin: value corresponding to minimum color in heatmap, default is 0.1
:type vmin: float, optional
:param vmax: value corresponding to minimum color in heatmap, default is 10^1
:type vmax: float, optional
:return: Sequence class enrichment dataframe. Rows correspond to risk loci sequence and the indices are the “SNP” column of <cite>ccv</cite> (the name/ID for each sequence). Columns correspond to the 40 sequence classes. 
:rtype: Pandas DataFrame</p>
</dd></dl>

<div class="toctree-wrapper compound">
</div>
</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="#">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">Welcome to SiFoN’s documentation!</a></li>
<li><a class="reference internal" href="#module-SiFoN.chrom_viz">chrom_viz API</a></li>
<li><a class="reference internal" href="#module-SiFoN.clinical_analysis">clinical_analysis API</a></li>
<li><a class="reference internal" href="#module-SiFoN.GWAS_enrichment">GWAS_enrichment API</a></li>
<li><a class="reference internal" href="#indices-and-tables">Indices and tables</a></li>
</ul>

  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/index.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="nav-item nav-item-0"><a href="#">SiFoN 0.0.1 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Welcome to SiFoN’s documentation!</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2022, Briana Macedo.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.4.0.
    </div>
  </body>
</html>