<!DOCTYPE html>
<html  lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1"><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

      <title>SiFoN.viz module</title>
    
          <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
          <link rel="stylesheet" href="_static/theme.css " type="text/css" />
      
      <!-- sphinx script_files -->
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>

      
      <!-- bundled in js (rollup iife) -->
      <!-- <script src="_static/theme-vendors.js"></script> -->
      <script src="_static/theme.js" defer></script>
    
  <link rel="index" title="Index" href="genindex.html" />
  <link rel="search" title="Search" href="search.html" />
  <link rel="next" title="SiFoN.chrom_viz module" href="SiFoN.chrom_viz.html" />
  <link rel="prev" title="SiFoN.GWAS_enrichment module" href="SiFoN.GWAS_enrichment.html" /> 
  </head>

  <body>
    <div id="app">
    <div class="theme-container" :class="pageClasses"><navbar @toggle-sidebar="toggleSidebar">
  <router-link to="index.html" class="home-link">
    
      <span class="site-name">SiFoN</span>
    
  </router-link>

  <div class="links">
    <navlinks class="can-hide">



    </navlinks>
  </div>
</navbar>

      
      <div class="sidebar-mask" @click="toggleSidebar(false)">
      </div>
        <sidebar @toggle-sidebar="toggleSidebar">
          
          <navlinks>
            



            
          </navlinks><div id="searchbox" class="searchbox" role="search">
  <div class="caption"><span class="caption-text">Quick search</span>
    <div class="searchformwrapper">
      <form class="search" action="search.html" method="get">
        <input type="text" name="q" />
        <input type="submit" value="Search" />
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div><div class="sidebar-links" role="navigation" aria-label="main navigation">
  
    <div class="sidebar-group">
      <p class="caption">
        <span class="caption-text"><a href="SiFoN.html#submodules">submodules</a></span>
      </p>
      <ul class="current">
        
          <li class="toctree-l1 ">
            
              <a href="SiFoN.GWAS_enrichment.html" class="reference internal ">SiFoN.GWAS_enrichment module</a>
            

            
          </li>

        
          <li class="toctree-l1 current">
            
              <a href="#" class="reference internal current">SiFoN.viz module</a>
            

            
              <ul>
                
                  <li class="toctree-l2"><a href="#module-SiFoN.viz" class="reference internal">SiFoN.viz module</a></li>
                
              </ul>
            
          </li>

        
          <li class="toctree-l1 ">
            
              <a href="SiFoN.chrom_viz.html" class="reference internal ">SiFoN.chrom_viz module</a>
            

            
          </li>

        
          <li class="toctree-l1 ">
            
              <a href="SiFoN.smoothing.html" class="reference internal ">SiFoN.smoothing module</a>
            

            
          </li>

        
          <li class="toctree-l1 ">
            
              <a href="SiFoN.clinical_analysis.html" class="reference internal ">SiFoN.clinical_analysis module</a>
            

            
          </li>

        
      </ul>
    </div>
  
</div>
        </sidebar>

      <page>
          <div class="body-header" role="navigation" aria-label="navigation">
  
  <ul class="breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
    
      <li><a href="SiFoN.html">SiFoN package</a> &raquo;</li>
    
    <li>SiFoN.viz module</li>
  </ul>
  

  <ul class="page-nav">
  <li class="prev">
    <a href="SiFoN.GWAS_enrichment.html"
       title="previous chapter">← SiFoN.GWAS_enrichment module</a>
  </li>
  <li class="next">
    <a href="SiFoN.chrom_viz.html"
       title="next chapter">SiFoN.chrom_viz module →</a>
  </li>
</ul>
  
</div>
<hr>
          <div class="content" role="main" v-pre>
            
  <section id="module-SiFoN.viz">
<span id="sifon-viz-module"></span><h1>SiFoN.viz module<a class="headerlink" href="#module-SiFoN.viz" title="Permalink to this headline">¶</a></h1>
<p>This module includes functions for visualizing sequence class scores along genomic regions. This module require Sei sequence class scores as input, and therefore assumes that predictions have already been calculated.</p>
<p>Functions:</p>
<blockquote>
<div><p>white_bg</p>
<p>preprocess</p>
<p>postprocess</p>
<p>find_max</p>
<p>find_max_by_category</p>
<p>plot_sequence_class</p>
<p>plot_max</p>
<p>plot_max_from_scores</p>
</div></blockquote>
<p>Misc variables:</p>
<blockquote>
<div><p>seqclass
color_map
class_map</p>
</div></blockquote>
<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.viz.find_max">
<span class="sig-prename descclassname"><span class="pre">SiFoN.viz.</span></span><span class="sig-name descname"><span class="pre">find_max</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">scores</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">signed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seqclass_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[]</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/SiFoN/viz.html#find_max"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#SiFoN.viz.find_max" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes the (abs) max sequence class score for each position/alteration.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>scores</strong> (<em>NumPy array</em>) – Takes the (abs) max sequence class score for each position/alteration. Should run the <cite>preprocessing</cite> function first. The output array has columns [“Max Score”, “Sequence Name”] corresponding the (abs) max score and the corresponding sequence class. This array is then past through the <cite>postprocess</cite> function.</p></li>
<li><p><strong>signed</strong> (<em>Boolean</em><em>, </em><em>optional</em>) – Signifies whether parameters are evaluated by absolute value (False) or not (True). This should be set to the same value as in the <cite>preprocessing</cite> function, default is True</p></li>
<li><p><strong>seqclass_names</strong> (<em>list of strings</em><em>, </em><em>optional</em>) – List of names for profiles. Sei sequence classes names or chromatin profile names. Should have length equal to the number of rows in`scores_max`, defaults to Sei sequence class names found in “model_data/seqclass-names.txt”</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Dataframe with one prediction per alteration, corresponding the maximum sequence class score and its corresponding sequence class name. Metadata used for graphing is also included in the dataframe.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Pandas DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.viz.find_max_by_category">
<span class="sig-prename descclassname"><span class="pre">SiFoN.viz.</span></span><span class="sig-name descname"><span class="pre">find_max_by_category</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">scores</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">signed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seqclass_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[]</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/SiFoN/viz.html#find_max_by_category"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#SiFoN.viz.find_max_by_category" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes the (abs) max sequence class score within a category for each position/alteration.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>scores</strong> (<em>NumPy array</em>) – Takes the (abs) max sequence class score within a category for each position/alteration. Should run the <cite>preprocessing</cite> function first. The output array has columns [“Max Score”, “Sequence Name”] corresponding the (abs) max score and the corresponding sequence class. This array is then past through the <cite>postprocess</cite> function.</p></li>
<li><p><strong>signed</strong> (<em>Boolean</em><em>, </em><em>optional</em>) – Signifies whether parameters are evaluated by absolute value (False) or not (True). This should be set to the same value as in the <cite>preprocessing</cite> function, default is True</p></li>
<li><p><strong>seqclass_names</strong> (<em>list of strings</em><em>, </em><em>optional</em>) – List of names for profiles. Sei sequence classes names or chromatin profile names. Should have length equal to the number of rows in`scores_max`, defaults to Sei sequence class names found in “model_data/seqclass-names.txt”</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Dataframe with seven predictions per alteration, corresponding to the maximum score within each of the seven sequence classes (excluding L and NA). Metadata used for graphing is also included in the dataframe.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Pandas DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.viz.plot_max">
<span class="sig-prename descclassname"><span class="pre">SiFoN.viz.</span></span><span class="sig-name descname"><span class="pre">plot_max</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filename</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scores_prune</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">TSS</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">yaxis_title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Max</span> <span class="pre">Score'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/SiFoN/viz.html#plot_max"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#SiFoN.viz.plot_max" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots the maximum Sei classes along a genomic sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>filename</strong> (<em>string</em>) – Name of file that figure will be saved as.</p></li>
<li><p><strong>scores_prune</strong> (<em>Pandas DataFrame</em>) – Dataframe that denotes the maximum Sei sequence classes along some region of the genome. Must contain the following columns: [“Max Score”, “Class”, “Position”, “Sequence Name”]. This is generated as output from the <cite>find_max</cite> and <cite>find_max_by_category</cite> functions.</p></li>
<li><p><strong>TSS</strong> (<em>Dict</em><em>[</em><em>str</em><em>, </em><em>list</em><em>[</em><em>int</em><em>, </em><em>int</em><em>]</em><em>]</em>) – Specifies regions of the genome to annotate in the figure. The first element of the dictionary (str) will be the name or label of the annotation which will appear in the legend. The list of ints will denote the end points of the region to be annotated.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.viz.plot_max_from_scores">
<span class="sig-prename descclassname"><span class="pre">SiFoN.viz.</span></span><span class="sig-name descname"><span class="pre">plot_max_from_scores</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filename</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vcf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">TSS</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">signed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_by</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'class'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pre</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seqclass_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[]</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/SiFoN/viz.html#plot_max_from_scores"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#SiFoN.viz.plot_max_from_scores" title="Permalink to this definition">¶</a></dt>
<dd><p>Preprocesses Sei prediction data, finds the maximum, and then plots the results. Runs <cite>preprocessing</cite>, <cite>find_max</cite> or <cite>find_max_by_category</cite>, and then <cite>plot_max</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>filename</strong> (<em>string</em>) – Name of file that figure will be saved as.</p></li>
<li><p><strong>file</strong> (<em>string</em>) – Filename of NumPy sequence class scores</p></li>
<li><p><strong>vcf</strong> (<em>Pandas DataFrame</em>) – VCF of SNPs of interest, including “Position” column.</p></li>
<li><p><strong>TSS</strong> (<em>Dict</em><em>[</em><em>str</em><em>, </em><em>list</em><em>[</em><em>int</em><em>, </em><em>int</em><em>]</em><em>]</em><em>, </em><em>optional</em>) – Specifies regions of the genome to annotate in the figure. The first element of the dictionary (str) will be the name or label of the annotation which will appear in the legend. The list of ints will denote the end points of the region to be annotated, defaults to an empty dictionary</p></li>
<li><p><strong>signed</strong> (<em>Boolean</em><em>, </em><em>optional</em>) – Signifies whether parameters are evaluated by absolute value (False) or not (True). This should be set to the same value as in the <cite>preprocessing</cite> function, default is True</p></li>
<li><p><strong>plot_by</strong> (<em>string</em><em>, </em><em>optional</em>) – Specifies whether the maximum of sequence class is plotted (“class”) or the maximim within each sequence category (“category). Should be either “class” (calls <cite>find_max</cite>) or “category” (calls <cite>find_max_by_category</cite>), defaults to “class”</p></li>
<li><p><strong>pre</strong> (<em>Boolean</em><em>, </em><em>optional</em>) – Determines whether to run preprocessing or not. If there is already only one alteration per position, then this should be set to False, default is False.</p></li>
<li><p><strong>seqclass_names</strong> (<em>list of strings</em><em>, </em><em>optional</em>) – List of names for profiles. Sei sequence classes names or chromatin profile names. Should have length equal to the number of rows in`scores_max`, defaults to Sei sequence class names found in “model_data/seqclass-names.txt”</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.viz.plot_sequence_class">
<span class="sig-prename descclassname"><span class="pre">SiFoN.viz.</span></span><span class="sig-name descname"><span class="pre">plot_sequence_class</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filename</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vcf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">TSS</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">category</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'All'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">signed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pre</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seqclass_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[]</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/SiFoN/viz.html#plot_sequence_class"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#SiFoN.viz.plot_sequence_class" title="Permalink to this definition">¶</a></dt>
<dd><p>Preprocesses Sei prediction data and then plots the results.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>filename</strong> (<em>string</em>) – Name of file that figure will be saved as.</p></li>
<li><p><strong>file</strong> (<em>string</em>) – Filename of NumPy sequence class scores</p></li>
<li><p><strong>vcf</strong> (<em>Pandas DataFrame</em>) – VCF of SNPs of interest, including “Position” column.</p></li>
<li><p><strong>TSS</strong> (<em>Dict</em><em>[</em><em>str</em><em>, </em><em>list</em><em>[</em><em>int</em><em>, </em><em>int</em><em>]</em><em>]</em><em>, </em><em>optional</em>) – Specifies regions of the genome to annotate in the figure. The first element of the dictionary (str) will be the name or label of the annotation which will appear in the legend. The list of ints will denote the end points of the region to be annotated, defaults to an empty dictionary</p></li>
<li><p><strong>category</strong> (<em>string</em><em>, </em><em>optional</em>) – Determines which sets of predictions will be plotted. Should be one of the category names listed in the <cite>color_map</cite>. If set to “All” then the full set of predictions is plotted, default is “All”</p></li>
<li><p><strong>signed</strong> (<em>Boolean</em><em>, </em><em>optional</em>) – Signifies whether parameters are evaluated by absolute value (False) or not (True). This should be set to the same value as in the <cite>preprocessing</cite> function, default is True</p></li>
<li><p><strong>pre</strong> (<em>Boolean</em><em>, </em><em>optional</em>) – Determines whether to run preprocessing or not. If there is already only one alteration per position, then this should be set to False, default is False.</p></li>
<li><p><strong>seqclass_names</strong> (<em>list of strings</em><em>, </em><em>optional</em>) – List of names for profiles. Sei sequence classes names or chromatin profile names. Should have length equal to the number of rows in`scores_max`, defaults to Sei sequence class names found in “model_data/seqclass-names.txt”</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.viz.postprocess">
<span class="sig-prename descclassname"><span class="pre">SiFoN.viz.</span></span><span class="sig-name descname"><span class="pre">postprocess</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">scores_max</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seqclass_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[]</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/SiFoN/viz.html#postprocess"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#SiFoN.viz.postprocess" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes the <cite>scores_max</cite> array to have information necessary for graphing, including sequence class category and color.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>scores_max</strong> – Type scores_max: NumPy Array</p></li>
<li><p><strong>seqclass_names</strong> (<em>list of strings</em><em>, </em><em>optional</em>) – List of names for profiles. Sei sequence classes names or chromatin profile names. Should have length equal to the number of rows in`scores_max`, defaults to Sei sequence class names found in “model_data/seqclass-names.txt”</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>an updated version of <cite>scores_max</cite> that adds the columns “Sequence Index”, “Class” (e..g E, P, TF), “Color” (corresponding to the “Class” column), “Function” (whether a sequence class is associated with repressed or active chromatin). SNPs with a max score in the Low Impact category are also removed.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>NumPy Array</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.viz.preprocess">
<span class="sig-prename descclassname"><span class="pre">SiFoN.viz.</span></span><span class="sig-name descname"><span class="pre">preprocess</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vcf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seqclass_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">signed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/SiFoN/viz.html#preprocess"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#SiFoN.viz.preprocess" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes data so that there is a single row per position representing the alteration with the maximum average score at that position.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>file</strong> (<em>string</em>) – Filename of NumPy sequence class scores</p></li>
<li><p><strong>vcf</strong> (<em>Pandas DataFrame</em>) – VCF of SNPs of interest, including “Position” column.</p></li>
<li><p><strong>seqclass_names</strong> (<em>list of strings</em><em>, </em><em>optional</em>) – List of names for profiles. Sei sequence classes names or chromatin profile names. Should have length equal to the number of rows in`scores_max`, defaults to Sei sequence class names found in “model_data/seqclass-names.txt”</p></li>
<li><p><strong>signed</strong> (<em>Boolean</em><em>, </em><em>optional</em>) – Signifies whether parameters are evaluated by absolute value (False) or not (True), default is True</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>array of scores, including a single row per position representing the alteration with the maximum average score</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>NumPy Array</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="SiFoN.viz.white_bg">
<span class="sig-prename descclassname"><span class="pre">SiFoN.viz.</span></span><span class="sig-name descname"><span class="pre">white_bg</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fig</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/SiFoN/viz.html#white_bg"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#SiFoN.viz.white_bg" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</section>


          </div>
          <div class="page-nav">
            <div class="inner"><ul class="page-nav">
  <li class="prev">
    <a href="SiFoN.GWAS_enrichment.html"
       title="previous chapter">← SiFoN.GWAS_enrichment module</a>
  </li>
  <li class="next">
    <a href="SiFoN.chrom_viz.html"
       title="next chapter">SiFoN.chrom_viz module →</a>
  </li>
</ul><div class="footer" role="contentinfo">
      &#169; Copyright 2022, Briana Macedo.
    <br>
    Created using <a href="http://sphinx-doc.org/">Sphinx</a> 4.4.0 with <a href="https://github.com/schettino72/sphinx_press_theme">Press Theme</a> 0.8.0.
</div>
            </div>
          </div>
      </page>
    </div></div>
    
    
  </body>
</html>