
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>SiFoN.GWAS_enrichment &#8212; SiFoN 0.0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/classic.css" />
    
    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="nav-item nav-item-0"><a href="../../index.html">SiFoN 0.0.1 documentation</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="../index.html" accesskey="U">Module code</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">SiFoN.GWAS_enrichment</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <h1>Source code for SiFoN.GWAS_enrichment</h1><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">matplotlib.colors</span> <span class="kn">import</span> <span class="n">LogNorm</span><span class="p">,</span> <span class="n">Normalize</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">pyranges</span> <span class="k">as</span> <span class="nn">pr</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>

<span class="k">def</span> <span class="nf">_sort_intersection</span><span class="p">(</span><span class="n">intersect_ref</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    intersect_ref :</span>
<span class="sd">        </span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>

<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">intersect_ref</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s2">&quot;End&quot;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;mergesort&#39;</span><span class="p">)</span>
    <span class="n">intersect_ref</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s2">&quot;Start&quot;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;mergesort&#39;</span> <span class="p">)</span>
    <span class="n">intersect_ref</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s2">&quot;Chromosome&quot;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;mergesort&#39;</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">intersect_ref</span>
    
<div class="viewcode-block" id="find_and_plot_enrichment"><a class="viewcode-back" href="../../SiFoN.GWAS_enrichment.html#SiFoN.GWAS_enrichment.find_and_plot_enrichment">[docs]</a><span class="k">def</span> <span class="nf">find_and_plot_enrichment</span><span class="p">(</span><span class="n">ccv</span><span class="p">,</span> <span class="n">ref_bed</span><span class="p">,</span> <span class="n">figname</span><span class="p">,</span> <span class="n">vmin</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="mi">10</span><span class="o">^</span><span class="mi">1</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;Calculates sequence class enrichment in genomic sequences compared to the entire genome and then creates a heatmap to visualize results.</span>

<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    ccv : Pandas DataFrame</span>
<span class="sd">        Pandas DataFrame in BED format of loci with credible causal variants. Must include at least the following columns: Chrom, Start, End, SNP [identifier/name of sequence])</span>
<span class="sd">    ref_bed : pyranges BED object</span>
<span class="sd">        Should contain sequence class labels across (nearly) the entire genome to allow for background sequence class proportion calculations. Should use the same genomic coordinate system (e.g. hg19 or hg38) as `ccv`.</span>
<span class="sd">    figname : string</span>
<span class="sd">        name of file that figure will be saves as.</span>
<span class="sd">    vmin : float, optional</span>
<span class="sd">        value corresponding to minimum color in heatmap, default is 0.1</span>
<span class="sd">    vmax : float, optional</span>
<span class="sd">        value corresponding to minimum color in heatmap, default is 10^1</span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    Pandas DataFrame</span>
<span class="sd">        Sequence class enrichment dataframe. Rows correspond to risk loci sequence and the indices are the &quot;SNP&quot; column of `ccv` (the name/ID for each sequence). Columns correspond to the 40 sequence classes.</span>

<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">intersect_bed</span> <span class="o">=</span> <span class="n">pr</span><span class="o">.</span><span class="n">PyRanges</span><span class="p">(</span><span class="n">ccv</span><span class="p">)</span>
    <span class="n">intersect_ref1</span> <span class="o">=</span> <span class="n">ref_bed</span><span class="o">.</span><span class="n">intersect</span><span class="p">(</span><span class="n">intersect_bed</span><span class="p">)</span><span class="o">.</span><span class="n">as_df</span><span class="p">()</span>
    <span class="n">intersect_ref2</span> <span class="o">=</span> <span class="n">_sort_intersection</span><span class="p">(</span><span class="n">intersect_ref1</span><span class="p">)</span>
    <span class="n">intersect_ref2</span> <span class="o">=</span> <span class="n">intersect_bed</span><span class="o">.</span><span class="n">intersect</span><span class="p">(</span><span class="n">ref_bed</span><span class="p">)</span><span class="o">.</span><span class="n">as_df</span><span class="p">()</span>
    <span class="n">intersect_ref2</span> <span class="o">=</span> <span class="n">_sort_intersection</span><span class="p">(</span><span class="n">intersect_ref2</span><span class="p">)</span>
    <span class="n">seqclass_names</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;../model_data/seqclass-names.txt&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">to_numpy</span><span class="p">()</span>
    <span class="n">seqclass_names</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">seqclass_names</span><span class="p">,</span> <span class="mi">21</span><span class="o">*</span><span class="p">[</span><span class="s2">&quot;NA&quot;</span><span class="p">])</span>
    <span class="n">intersect_ref2</span><span class="p">[</span><span class="s2">&quot;Sequence Identity&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">intersect_ref1</span><span class="p">[</span><span class="s2">&quot;Name&quot;</span><span class="p">]</span>
    <span class="n">intersect_ref2</span><span class="p">[</span><span class="s1">&#39;Sei Sequence Annotation&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">seqclass_names</span><span class="p">[</span><span class="n">name</span><span class="p">]</span> <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">intersect_ref2</span><span class="p">[</span><span class="s2">&quot;Sequence Identity&quot;</span><span class="p">]]</span>
    
    <span class="n">ref</span> <span class="o">=</span> <span class="n">ref_bed</span><span class="o">.</span><span class="n">as_df</span><span class="p">()</span> <span class="c1"># proportions of counts across the genome</span>
    <span class="n">ref</span><span class="p">[</span><span class="s1">&#39;Sei Sequence Annotation&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">seqclass_names</span><span class="p">[</span><span class="n">name</span><span class="p">]</span> <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">ref</span><span class="p">[</span><span class="s2">&quot;Name&quot;</span><span class="p">]]</span>
    <span class="n">total_counts</span> <span class="o">=</span> <span class="n">ref</span><span class="p">[</span><span class="s1">&#39;Sei Sequence Annotation&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span><span class="o">.</span><span class="n">rename_axis</span><span class="p">(</span><span class="s1">&#39;Sei Sequence Annotation&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s1">&#39;counts&#39;</span><span class="p">)</span>
    <span class="n">total_counts</span><span class="p">[</span><span class="s2">&quot;counts&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">total_counts</span><span class="p">[</span><span class="s2">&quot;counts&quot;</span><span class="p">]</span><span class="o">/</span><span class="nb">sum</span><span class="p">(</span><span class="n">total_counts</span><span class="p">[</span><span class="s2">&quot;counts&quot;</span><span class="p">])</span>
    <span class="n">total_counts</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Sei Sequence Annotation&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="n">counts_in_loci</span> <span class="o">=</span> <span class="n">intersect_ref2</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;SNP&quot;</span><span class="p">,</span><span class="s1">&#39;Sei Sequence Annotation&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">size</span><span class="p">()</span><span class="o">.</span><span class="n">unstack</span><span class="p">(</span><span class="n">fill_value</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="n">counts_in_loci</span><span class="o">.</span><span class="n">sort_index</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="n">counts_in_loci</span> <span class="o">=</span> <span class="n">counts_in_loci</span><span class="o">.</span><span class="n">div</span><span class="p">(</span><span class="n">counts_in_loci</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>  <span class="c1"># calculate proportion of each label per region</span>
    <span class="n">counts_in_loci</span> <span class="o">=</span> <span class="n">counts_in_loci</span><span class="o">.</span><span class="n">div</span><span class="p">(</span><span class="n">total_counts</span><span class="p">[</span><span class="s2">&quot;counts&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span> <span class="c1"># normalize by proportion across the full genome</span>
    <span class="n">counts_in_loci</span> <span class="o">=</span> <span class="n">counts_in_loci</span> <span class="o">+</span> <span class="mf">0.00000001</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">rc</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;figure.figsize&#39;</span><span class="p">:(</span><span class="mi">24</span><span class="p">,</span><span class="mi">16</span><span class="p">)})</span>
    <span class="n">cmap</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">diverging_palette</span><span class="p">(</span><span class="mi">255</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="mi">8</span><span class="p">,</span> <span class="n">n</span><span class="o">=</span><span class="mi">256</span><span class="p">)</span>
    <span class="n">g</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">clustermap</span><span class="p">(</span><span class="n">counts_in_loci</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span> 
                   <span class="n">vmin</span><span class="o">=</span><span class="n">vmin</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="n">vmax</span><span class="p">,</span> <span class="n">xticklabels</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>  <span class="n">norm</span><span class="o">=</span><span class="n">LogNorm</span><span class="p">())</span>
    <span class="n">g</span><span class="o">.</span><span class="n">ax_col_dendrogram</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">([</span><span class="mi">0</span><span class="p">,</span><span class="mf">0.0001</span><span class="p">])</span>
    <span class="n">g</span><span class="o">.</span><span class="n">ax_row_dendrogram</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">g</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="n">figname</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
    
    <span class="k">return</span> <span class="n">counts_in_loci</span></div>
</pre></div>

            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="nav-item nav-item-0"><a href="../../index.html">SiFoN 0.0.1 documentation</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="../index.html" >Module code</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">SiFoN.GWAS_enrichment</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2022, Briana Macedo.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.4.0.
    </div>
  </body>
</html>